### PR TITLE
Adapt to coq/coq#19927 (coq-core renamed to rocq-runtime)

### PR DESCRIPTION
**Pull request by SkySkimmer**
There are many more occurrences of `coq-core` in this repo but I don't know what they're doing. This patch is just enough to fix Coq's CI.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
Doesn't quite work, I get 
~~~diff
-   - 2 Coq path directory bindings in scope; 27 Coq plugin directory bindings in scope
+   - 2 Coq path directory bindings in scope; 0 Coq plugin directory bindings in scope
~~~
eg https://gitlab.inria.fr/coq/coq/-/jobs/5062976

**Comment by ejgallego** [comment]
I'll have a look tomorrow and try to fix it.

**Comment by ejgallego** [comment]
@SkySkimmer you missed the occurence in `coq/args.ml` which the important one.

Anyways this code can be removed as we don't use the findlib method anymore, I actually have a coq-lsp branch that does that but it will take a while for me to return back to work unfortunately.

**Comment by ejgallego** [comment]
@SkySkimmer you mind bumping the `vendor/coq` submodule to I can use coq-lsp CI to check things are right?

(And adapt to other changes like the worker build, etc...)

Also serlib needs (an easy) fix for its `dune` files, that basically covers it all.

Note that I plan to make the loader not so hacky, also got a branch for that somewhere.

**Comment by SkySkimmer** [comment]
>@SkySkimmer you mind bumping the vendor/coq submodule to I can use coq-lsp CI to check things are right?

IDK how to do that

**Comment by SkySkimmer** [comment]
But the vendored git repo only knows about coq/coq, my branch is in skyskimmer/coq

**Comment by ejgallego** [comment]
Good questions, I'm not sure I make it work, as I use `hub` and `gh` command line tools. Maybe `git submodule set-url` is the new recommended way?

Never mind if that's too hard.

**Comment by SkySkimmer** [comment]
I edited .gitmodules to point at my fork

**Comment by ejgallego** [comment]
Updated rest of the system, including the missing change to `coq/args` that made Coq CI fail.

Let's see how it goes.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -170,7 +170,7 @@ jobs:
         run: |
           opam install lwt logs # Also build pet-server
           opam install memprof-limits # We need to do this to avoid coq-lsp rebuilding Coq below due to deptops
-          opam install vendor/coq/{coq-core,rocq-core,stdlib/coq-stdlib,coqide-server,coq}.opam
+          opam install vendor/coq/{rocq-runtime,rocq-core,stdlib/coq-stdlib,coqide-server,coq}.opam
```
Comment (SkySkimmer) [inline .github/workflows/build.yml:173]: stdlib still depends on coq-core until a followup PR
and coq will depend on coq-core forever (but we will add a `rocq` package ofc)
so just removing coq-core from the "to install" list probably won't work

Comment (ejgallego) [reply .github/workflows/build.yml:173]: Rest is working fine, waiting for the web worker artifact to test it.

**Comment by ejgallego** [comment]
>  - 2 Coq path directory bindings in scope; 0 Coq plugin directory bindings in scope

As I've mentioned, this will change to "X findlib paths in scope"

**Review by ejgallego — APPROVED** [review]
CI green, local and JS tests OK too.

**Pull request merged**